src/runtime/rmisc.r

```diff
@@ -1843,7 +1843,7 @@ int pattern_image(union block *pe, int prev_index, dptr result,
                 image_case = PT_VP;
              else if(Blk(ep, Pelem)->pcode == PC_BreakX_MF)
                 image_case = PT_MF;
-             else if(Blk(ep, Pelem)->pcode == PC_BreakX_CS)
+             else /* if(Blk(ep, Pelem)->pcode == PC_BreakX_CS)  <-- must always be true */
```

Comment:
I don't get this.

Reply:
Here is an extract from our email conversation (Oct 2020). 

rmisc.r:1846:22: warning: variable 'image_case' is used uninitialized whenever 'if' condition  is false [-Wsometimes-uninitialized]
          case PC_BreakX_VF:
          case PC_BreakX_VP:
          case PC_BreakX_MF:
          case PC_BreakX_CS: {
             int image_case; 

             if (Blk(ep, Pelem)->pcode == PC_BreakX_VF)
                image_case = PT_VF;
             else if(Blk(ep, Pelem)->pcode == PC_BreakX_VP)
                image_case = PT_VP;
             else if(Blk(ep, Pelem)->pcode == PC_BreakX_MF)
                image_case = PT_MF;
             else if(Blk(ep, Pelem)->pcode == PC_BreakX_CS)
                image_case = PT_EVAL; 

             if (construct_funcimage(ep, image_case, Blk(ep, Pelem)->pcode, 
                             result, index_image) == RunError) return RunError;

I think clang is just being dim here. The “offending” code is inside a switch statement
      switch (Blk(ep,Pelem)->pcode) {
So, given the four case statements, image_case can never end up not being initialised. We can either suppress the warning or, perhaps, omit the final if statement like this
             else /* if(Blk(ep, Pelem)->pcode == PC_BreakX_CS) */
                image_case = PT_EVAL; 
Or we can swallow our pride and initialise the variable, even if it doesn’t actually need to be initialised.
The final possibility is 
             int image_case = (Blk(ep, Pelem)->pcode == PC_BreakX_VF)
                               ? PT_VF 
                               : ((Blk(ep, Pelem)->pcode == PC_BreakX_VP)
                                   ? PT_VP
                                   : ... etc.
But I don’t think that’s as clear as the present code.